src/main/kotlin/biokotlin/kmer/KmerIO.kt

```diff
@@ -0,0 +1,225 @@
+package biokotlin.kmer
+
+import it.unimi.dsi.fastutil.BigArrays
+import net.jpountz.lz4.LZ4FrameInputStream
+import net.jpountz.lz4.LZ4FrameOutputStream
+import java.io.*
+
+/**
+ * This class reads kmers from a text file (which may be LZ4 compressed)
+ * and can be used to construct KmerSet, KmerBigSet, and KmerMultiSet objects.
+ * It can also iterate through a kmer text file without constructing a set.
+ */
+class KmerIO(filename: String, isCompressed: Boolean = true): Iterator<Pair<Kmer, Int>> {
+    private val reader: BufferedReader
+
+    private var sequenceLength: Long? = null
```

Comment:
Is there a way to reduce the nullability with these variables?  Do they have reasonable default values that if they are that you could still throw the warning messages?